code/modules/projectiles/guns/energy/stun.dm

```diff
@@ -66,6 +66,28 @@
 	projectile_type = "/obj/item/projectile/energy/electrode"
 	cell_type = "/obj/item/weapon/cell"
 
+/obj/item/weapon/gun/energy/stunrevolver/failure_check(var/mob/living/carbon/human/M)
+	if(damaged)
+		if(prob(15))
+			fire_delay += 2
+			to_chat(M, "<span class='warning'>\The [src] buzzes.</span>")
+			return 1
+		if(prob(5))
```

Comment:
Yes, in theory. Which is why a 1 in 20 chance of effectively leaving its owner wide open to certain death is bad.